### PR TITLE
feat(INVT-CPC-1150): Edit Supplies Inside An Inventory

### PR DESCRIPTION
**Pull request by Sunveerg**
[**JIRA:** link to jira ticket](https://champlainsaintlambert.atlassian.net/browse/CPC-1150)
## Context:
This push is for the front end and back end of the edit inventory products. Also, it includes negative and positive testing for the new methods in the Inventory Controller in the API-Gateway 
## Does this PR change the .vscode folder in petclinic-frontend?:
No

## Changes

- Added edit button
- When you click the button, it redirects you to a new page
- The new page has a form pre-populated with the information on the inventory product 
- You can edit the form and save your changes

## Before and After UI (Required for UI-impacting PRs)
Before
image
After
image
image

## Dev notes (Optional)
Please accept my Pull Request

### CODE REVIEW

File: petclinic-frontend/src/features/inventories/EditInventoryProducts.tsx [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    }
+    if (!product.productQuantity) {
+      newError.productQuantity = 'Product quantity is required';
+    }
+    if (!product.productSalePrice) {
+      newError.productSalePrice = 'Product sale price is required';
+    }
+    setError(newError);
+    return Object.keys(newError).length === 0;
+  };
```
Comment (CharlesS42) [inline petclinic-frontend/src/features/inventories/EditInventoryProducts.tsx:72]: Good idea setting a validation of input mechanism. Later, we could also implement a type validation as well for product quantity for example.

**Review by CharlesS42 — APPROVED** [review]

File: petclinic-frontend/src/features/inventories/EditInventoryProducts.tsx (hunk 1 above)
Comment (Sunveerg) [reply petclinic-frontend/src/features/inventories/EditInventoryProducts.tsx:72]: Great idea

File: petclinic-frontend/src/features/inventories/EditInventoryProducts.tsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+          <input
+            type="number"
+            name="productQuantity"
+            value={product.productQuantity}
+            onChange={handleChange}
+            required
+          />
+        </div>
+
+        <div>
```
Comment (vinivelozo) [inline petclinic-frontend/src/features/inventories/EditInventoryProducts.tsx:163]: Great but I tkink we don;t use the sale price anymore maybe change this later

**Review by vinivelozo — APPROVED** [review]
Great changes, no changes on the vscode and maybe just take out the sale price in the future since we are not using it on the front end.

File: petclinic-frontend/src/features/inventories/EditInventoryProducts.tsx (hunk 2 above)
Comment (Sunveerg) [reply petclinic-frontend/src/features/inventories/EditInventoryProducts.tsx:163]: Ok noted thanks

**Pull request merged**